### PR TITLE
docs update w.r.t. dependencies vs dummy toolchain, downloading over proxy, clean_gists.py script

### PR DESCRIPTION
**Pull request by boegel**
includes reviewed version of #78 by @pescobar and  #96 by @wpoely86

@pescobar, @wpoely86, @fgeorgatos: please review?

rendered versions of the touched sections:
* http://boegel-eb.readthedocs.org/en/download_over_proxy/Writing_easyconfig_files.html#source-files-and-patches
* http://boegel-eb.readthedocs.org/en/download_over_proxy/Writing_easyconfig_files.html#loading-of-modules-for-dependencies-with-a-dummy-toolchain
* http://boegel-eb.readthedocs.org/en/download_over_proxy/Writing_easyconfig_files.html#specifying-dependencies-using-dummy-toolchain
* http://boegel-eb.readthedocs.org/en/download_over_proxy/Useful-scripts.html#clean-gists-py

### CODE REVIEW

File: docs/Useful-scripts.rst [hunk 1]
```diff
@@ -81,3 +81,24 @@ This results in ``broken.eb`` being rewritten/fixed, after creating a backup cop
...
+
+To help with that the ``clean_gists.py`` script is available, to clean up gists containing test reports:
+
+* ``clean_gists.py -p``: delete all gists from closed pull requests (default action if no other action is specified)
+* ``clean_gists.py -a``: delete all gists generated by Easybuild
+* ``clean_gists.py -o``: delete all gists without a matching pull request (created by using ``-upload-test-report`` without ``--from-pr``)
+
+By default, the script will use the same GitHub account that Easybuild uses (see ``--github-user``); to specify a different GitHub account, use ``-g``.
+
+The script expects that a valid GitHub token for the used GitHub account username is available, see ref:`github_token`.
```
Comment (wpoely86) [inline docs/Useful-scripts.rst]: this ref points to nothing?

File: docs/Writing_easyconfig_files.rst [hunk 2]
```diff
@@ -25,8 +25,8 @@ It consists of a plain text file (in Python syntax) with mostly `key-value` assi
 
 Easyconfigs typically follow a (fixed) strict naming scheme, i.e.  ``<name>-<version>[-<toolchain>][-<versionsuffix>].eb``.
 
-The toolchain label (which includes the toolchain name and version) is omitted when a ``dummy`` toolchain is used
-(see also :ref:`dummy_toolchain`); the version suffix is omitted when it's empty.
+The toolchain label (which includes the toolchain name and version) is omitted when a :ref:`dummy_toolchain` is used;
+the version suffix is omitted when it's empty.
```
Comment (wpoely86) [inline docs/Writing_easyconfig_files.rst]: what is empty?

Comment (boegel) [inline docs/Writing_easyconfig_files.rst]: the version suffix... ;-)

File: docs/Useful-scripts.rst (hunk 1 above)
Comment (boegel) [inline docs/Useful-scripts.rst]: hmm, weird... seems like cross-references labels across pages isn't working like I thought it was

Comment (boegel) [inline docs/Useful-scripts.rst]: ah, should be `:ref:`...

**Comment by fgeorgatos** [comment]
reviewed, all ok

**Comment by boegel** [comment]
Thanks for the review @fgeorgatos, @wpoely86!

**Pull request merged**